core/src/bmi.jl

```diff
@@ -223,6 +229,13 @@ function create_callbacks(
     tabulated_rating_curve_cb = PresetTimeCallback(tstops, update_tabulated_rating_curve!)
     push!(callbacks, tabulated_rating_curve_cb)
 
+    if config.allocation.use_allocation
+        t_end = seconds_since(config.endtime, config.starttime)
+        tstops = 1:(config.allocation.timestep):t_end
+        allocation_cb = PresetTimeCallback(tstops, update_allocation!)
```

Comment:
For a fixed interval the PeriodicCallback is a better match: https://docs.sciml.ai/DiffEqCallbacks/stable/timed_callbacks/